src/test/java/com/puppycrawl/tools/checkstyle/AbstractPathTestSupport.java

```diff
@@ -27,9 +27,15 @@
 
 public abstract class AbstractPathTestSupport {
 
-    protected static final String LF_REGEX = "\\\\n";
-
-    protected static final String CRLF_REGEX = "\\\\r\\\\n";
+    // we are using positive lookahead here, to convert \r\n to \n
+    // and \\r\\n to \\n (for parse tree dump files),
+    // by replacing the full match with the empty string
+    private static final String CR_FOLLOWED_BY_LF_REGEX = "(?x)"
+            + "\\\\r     # BACKSLASH_R \n"
+            + "(?=\\\\n) # followed by BACKSLASH_N (not part of the match) \n"
```

Comment:
I assume `#` are comments. Why are they embedded in the string and not a true java comment?